index.js

```diff
@@ -1,37 +1,57 @@
 const express = require("express");
-const fs = require('fs');
-const { Worker } = require('node:worker_threads');
-const path = require('path');
+const fs = require("fs");
+const { Worker } = require("node:worker_threads");
+const path = require("path");
+const crypto = require("crypto");
 
 var cors = require("cors");
-var bodyParser = require('body-parser');
 
 const app = express();
 
-const port = 4000;
+app.use(cors());
+app.use(express.json());
+
+// TODO: remove logger middleware
+app.use((req, res, next) => {
+  req.time = new Date(Date.now()).toString();
+  console.log(req.method, req.hostname, req.path, req.time);
+  console.log(req.body);
+  console.log(users);
+  next();
+});
 
+const PORT = 4000;
 const USER_COUNT_FETCH_INTERVAL = 15 * 60 * 1000;
 
-const names = [];
+//TODO: persist this
+//TODO: check lower case names?
+const users = new Map();
```

Comment:
I think it was changed to a Map so it can take additional values, not just a name string. See line 43.
So the same functionality is there, just names being used as the key of this map. See line 34.